### ISSUE TITLE
Configurable rules are not working in nested `.yml` file

### ISSUE THREAD

**Issue by aleksandergrzyb** [realm/SwiftLint#2662]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

We run `swiftlint` as a pre-commit hook in the root of our repository. We have one `.yml` file placed in `Folder1/Folder2/.swiftlint.yml` path. When change is made to a file stored in `Folder1/Folder2` path, configurable rules (such us `line_length`) are not working. For example we have warning set to 150 characters for `line_length`, but SwiftLint returns violation `warning: Line Length Violation: Line should be 120 characters or less: currently 135 characters (line_length)`. It looks like SwiftLint ignores this setting.

##### Complete output when running SwiftLint, including the stack trace and command used

Command:

```bash
$ SCRIPT_INPUT_FILE_COUNT=1 SCRIPT_INPUT_FILE_0=Folder1/Folder2/File.swift ./Toolset/SwiftLint/swiftlint-0.30.1 lint --strict --use-script-input-files
```

Output:

```bash
Done linting! Found 1 violation, 0 serious in 1 file.
Folder1/Folder2/File.swift :47:1: warning: Line Length Violation: Line should be 120 characters or less: currently 135 characters (line_length)
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?

0.30.1

* Installation method used (Homebrew, CocoaPods, building from source, etc)?

Binary stored in repository.

* Paste your configuration file:

```yml
excluded: # paths to ignore during linting. overridden by `included`.
  - Carthage
  - Pods

disabled_rules:
  - nesting
  - identifier_name
  - type_name
  - xctfail_message

closing_brace: error
comma: error
colon:
  severity: error
control_statement: error
cyclomatic_complexity:
  - 10 # two nested ifs are acceptable
  - 12 # ten causes compile error
leading_whitespace: error
legacy_cggeometry_functions: error
legacy_constructor: error
legacy_nsgeometry_functions: error
line_length:
  warning: 150
  error: 150
mark: error
opening_brace: error
operator_whitespace: error
return_arrow_whitespace: error
statement_position:
  severity: error
  statement_mode: default
trailing_newline: error
trailing_whitespace:
  ignores_empty_lines: false
  severity: error
type_body_length:
  - 300 # warning
  - 400 # error
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?

We have one `.swiftlint.yml` file placed in `Folder1/Folder2/` path.

* Which Xcode version are you using (check `xcode-select -p`)?

10.1

* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

`swiftlint lint --path Folder1/Folder2/File.swift --no-cache`

`Folder1/Folder2/File.swift:47: warning: Line Length Violation: Line should be 120 characters or less: currently 135 characters (line_length)`

**Comment by aleksandergrzyb** [comment]
Ok, I solved the problem by passing relative path to repository, not a global one. 👍
